share/models/sources.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+from django.db import models
+
+from share.models.ingest import SourceConfig
+from share.models.ingest import NaturalKeyManager
+
+logger = logging.getLogger(__name__)
+__all__ = ('SourceStat',)
+
+
+class SourceStat(models.Model):
+    config = models.ForeignKey(SourceConfig, on_delete=models.CASCADE)
+    is_deleted = models.BooleanField(default=False)
+    date_created = models.DateTimeField(auto_now_add=True)
+    response_status_code = models.SmallIntegerField(blank=True, null=True)
+    response_elapsed_time = models.FloatField(blank=True, null=True)
+    response_exception = models.TextField(blank=True, null=True)
+    earliest_datestamp_config = models.DateField(blank=True, null=True)
+    base_url_config = models.TextField()
+    admin_note = models.TextField(blank=True)
+
+    # OAI specific
+    earliest_datestamp_source = models.DateField(blank=True, null=True)
+    earliest_datestamps_match = models.BooleanField(default=False)
+
+    base_url_source = models.TextField(blank=True, null=True)
+    base_urls_match = models.BooleanField(default=False)
+
+    objects = NaturalKeyManager('config.label')
```

Comment:
Yeah, natural keys should be unique. If `config` were a one-to-one field, this would work (except it should be `config__label`). But it looks like there could be many SourceStats per SourceConfig?